src/datapilot/core/platforms/dbt/insights/checks/check_column_name_contract.py

```diff
@@ -0,0 +1,152 @@
+import re
+from typing import ClassVar
+from typing import List
+from typing import Sequence
+from typing import Tuple
+
+from datapilot.config.utils import get_check_config
+from datapilot.core.insights.utils import get_severity
+from datapilot.core.platforms.dbt.insights.checks.base import ChecksInsight
+from datapilot.core.platforms.dbt.insights.schema import DBTInsightResult
+from datapilot.core.platforms.dbt.insights.schema import DBTModelInsightResponse
+from datapilot.core.platforms.dbt.schemas.manifest import AltimateResourceType
+from datapilot.core.platforms.dbt.wrappers.catalog.wrapper import BaseCatalogWrapper
+from datapilot.utils.formatting.utils import numbered_list
+
+
+class CheckColumnNameContract(ChecksInsight):
+    NAME = "Column name follows contract pattern"
+    ALIAS = "column_name_contract"
+    DESCRIPTION = "Column names should adhere to the contract pattern defined for the data type. "
+    REASON_TO_FLAG = (
+        "Column names that do not adhere to the contract can lead to confusion and hinder effective data "
+        "modeling and analysis. It's important to maintain consistent column naming conventions."
+    )
+    FAILURE_MESSAGE = (
+        "The following columns in the model `{model_unique_id}` do not adhere to the contract:\n{columns}. "
+        "Inconsistent column naming conventions can impede understanding and usage of the model."
+    )
+    RECOMMENDATION = (
+        "Update the column names listed above in the model `{model_unique_id}` to adhere to the contract. "
+        "Consistent column naming conventions provide valuable context and aids in data understanding and collaboration."
+    )
+    PATTERN_STR = "pattern"
+    DATATYPE_STR = "dtype"
+    PATTERNS_LIST_STR = "patterns"
+    DEFAULT_PATTERN_STR = "default_pattern"
+    FILES_REQUIRED: ClassVar = ["Manifest", "Catalog"]
+
+    def __init__(self, catalog_wrapper: BaseCatalogWrapper, *args, **kwargs):
+        self.catalog = catalog_wrapper
+        super().__init__(*args, **kwargs)
+
+    def generate(self, *args, **kwargs) -> List[DBTModelInsightResponse]:
+        self.default_pattern = get_check_config(self.config, self.ALIAS, self.DEFAULT_PATTERN_STR)
```

Comment:
shouldn't get_check_config be an instance method?

Reply:
This is a common method for fetching required information about the config from config,yaml. We could have it in the base class